tt_metal/common/work_split.cpp

```diff
@@ -418,5 +418,47 @@ std::tuple<uint32_t, CoreRangeSet, CoreRangeSet, CoreRangeSet, uint32_t, uint32_
         target_num_cores, all_cores, core_group_1, core_group_2, units_per_core_group_1, units_per_core_group_2);
 }
 
+std::tuple<uint32_t, CoreRangeSet, CoreRangeSet, CoreRangeSet, uint32_t, uint32_t> split_work_to_cores_aligned(
+    const CoreCoord grid_size, const uint32_t units_to_divide, const uint32_t alignment) {
```

Comment:
const reference for CoreCoord?

can the existing work split this if the units of work = align(tensor_side, alignment)? Mostly just curious about whether or not you could have used the existing one.